Modules/selectmodule.c

```diff
@@ -532,11 +532,12 @@ poll_poll(pollObject *self, PyObject *args)
         return NULL;
     }
 
+    deadline = 0;   /* initialize to prevent gcc warning */
+
     /* Check values for timeout */
     if (timeout_obj == NULL || timeout_obj == Py_None) {
         timeout = -1;
         ms = -1;
-        deadline = 0;   /* initialize to prevent gcc warning */
```

Comment:
See my comment about `INFTIM` below. I think we don't need to move this line out of this `if` branch.